src/lib/mina.ts

```diff
@@ -163,69 +205,151 @@ export function LocalBlockchain({
+          nonce: new UInt32(ledgerAccount.nonce.value),
+          zkapp: ledgerAccount.zkapp,
+        };
+      }
+    },
+    sendTransaction(txn: Transaction) {
+      txn.sign();
+      ledger.applyPartiesTransaction(
+        toParties(txn.transaction),
+        accountCreationFee_
+      );
+      return { wait: async () => {} };
+    },
+    async transaction(sender: SenderSpec, f: () => void) {
+      return createTransaction(sender, f);
+    },
+    applyJsonTransaction(json: string) {
+      return ledger.applyJsonTransaction(json, accountCreationFee_);
+    },
+    addAccount,
+    testAccounts,
+  };
+}
 
+function RemoteBlockchain(
+  graphqlEndpoint: string,
+  testAccounts = [] as {
+    publicKey: PublicKey;
+    privateKey: PrivateKey;
+  }[]
+): Testnet {
+  let accountCreationFee = UInt32.fromNumber(defaultAccountCreationFee);
+  Fetch.setGraphqlEndpoint(graphqlEndpoint);
   return {
-    accountCreationFee: () => UInt32.fromString(accountCreationFee_),
-    currentSlot,
-    getAccount,
-    transaction,
-    applyJsonTransaction,
-    addAccount,
+    accountCreationFee: () => accountCreationFee,
+    currentSlot() {
+      throw Error(
+        'currentSlot() is not implemented yet for remote blockchains.'
+      );
+    },
+    getAccount(publicKey: PublicKey) {
+      if (currentTransaction?.fetchMode === 'test') {
+        Fetch.markAccountToBeFetched(publicKey, graphqlEndpoint);
+        return dummyAccount(publicKey);
+      }
+      if (
+        currentTransaction == undefined ||
+        currentTransaction.fetchMode === 'cached'
+      ) {
+        let account = Fetch.getCachedAccount(publicKey, graphqlEndpoint);
+        if (account !== undefined) return account;
+      }
+      throw Error(
+        `getAccount: Could not find account for public key ${publicKey.toBase58()}.\nGraphql endpoint: ${graphqlEndpoint}`
+      );
+    },
+    sendTransaction(txn: Transaction) {
+      txn.sign();
+      let sendPromise = Fetch.sendZkapp(txn.toJSON());
+      return {
+        async wait() {
+          let [response, error] = await sendPromise;
+          console.log('got graphql response', { response, error });
+          console.log(
+            'Info: waiting for inclusion in a block is not implemented yet.'
+          );
+          // if (error !== undefined) {
+          //   console.log('Graphql transaction failed. Query:');
+          //   console.log(txn.toGraphqlQuery());
+          // }
+        },
+      };
+    },
+    async transaction(sender: SenderSpec, f: () => void) {
+      createTransaction(sender, f, { fetchMode: 'test' });
+      await Fetch.fetchMissingAccounts(graphqlEndpoint);
+      return createTransaction(sender, f, { fetchMode: 'cached' });
+    },
     testAccounts,
   };
 }
 
+function BerkeleyQANet(graphqlEndpoint: string): Testnet {
+  let whaleKey = PrivateKey.fromBase58(
+    'EKEtDLmauasBxaB3FsPVLHBdFHUoQMYa21y8phQwB9dV1fL4M2Cb'
```

Comment:
this key is pwned now as well

Reply:
should I remove it? I'm worried about losing the smooth testing flow this enables

Reply:
removed the testAccounts, including this key, now